File: WinForms/Config/VideoConfig.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CursedSodiumConfig.WinForms.Config
{
    internal class VideoConfig : ICloneable
    {
        // General Options
        public int RenderDistance { get; set; } = 12;
        public int SimulationDistance { get; set; } = 12;
        public int Brightness { get; set; } = 50;

        public int GuiScale { get; set; } = 0;
        public bool Fullscreen { get; set; } = false;
        public bool VSync { get; set; } = true;
        public int MaxFramerate { get; set; } = 120;

        public bool ViewBobbing { get; set; } = true;
        public AttackIndicator AttackIndicator { get; set; } = AttackIndicator.Crosshair;
        public bool AutosaveIndicator { get; set; } = true;

        // Quality Options
        public GraphicsMode Graphics { get; set; } = GraphicsMode.Fancy;

        public bool Clouds { get; set; } = true;
        public GraphicsQuality Weather { get; set; } = GraphicsQuality.Default;
        public GraphicsQuality Leaves { get; set; } = GraphicsQuality.Default;
        public ParticlesMode Particles { get; set; } = ParticlesMode.All;
        public bool SmoothLighting { get; set; } = true;
        public int BiomeBlend { get; set; } = 2;
        public int EntityDistance { get; set; } = 100;
        public bool EntityShadows { get; set; } = true;
        public bool Vignette { get; set; } = true;

        public int MipmapLevels { get; set; } = 2;

        // Performance Options
        public int ChunkUpdateThreads { get; set; } = 0;
        public bool AlwaysDeferChunkUpdates { get; set; } = true;

        public bool UseBlockFaceCulling { get; set; } = true;
        public bool UseFogOccusion { get; set; } = true;
        public bool UseEntityCulling { get; set; } = true;
        public bool AnimateOnlyVisibleTextures { get; set; } = true;
        public bool UseNoErrorContext { get; set; } = true;

        // Advanced Options
        public bool UsePersistentMapping { get; set; } = true;
        public int CpuRenderAhendLimit { get; set; } = 3;

        public VideoConfig()
        {
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}

File: src/main/java/io/grayray75/mods/cursedsodiumconfig/config/VideoConfig.java
package io.grayray75.mods.cursedsodiumconfig.config;

public class VideoConfig {
    // General Options
    public int RenderDistance;
    public int SimulationDistance;
    public int Brightness;

    public int GuiScale;
    public boolean Fullscreen;
    public boolean VSync;
    public int MaxFramerate;

    public boolean ViewBobbing;
    public int AttackIndicator;
    public boolean AutosaveIndicator;

    // Quality Options
    public int Graphics;

    public boolean Clouds;
    public int Weather;
    public int Leaves;
    public int Particles;
    public boolean SmoothLighting;
    public int BiomeBlend;
    public int EntityDistance;
    public boolean EntityShadows;
    public boolean Vignette;

    public int MipmapLevels;

    // Performance Options
    public int ChunkUpdateThreads;
    public boolean AlwaysDeferChunkUpdates;

    public boolean UseBlockFaceCulling;
    public boolean UseFogOccusion;
    public boolean UseEntityCulling;
    public boolean AnimateOnlyVisibleTextures;
    public boolean UseNoErrorContext;

    // Advanced Options
    public boolean UsePersistentMapping;
    public int CpuRenderAhendLimit;
}
